knip.ts

```diff
@@ -0,0 +1,136 @@
+import type { KnipConfig } from 'knip';
+
+export default {
+  rules: {
+    classMembers: 'off',
+    duplicates: 'off',
+    enumMembers: 'off',
+    exports: 'off',
+    nsExports: 'off',
+    nsTypes: 'off',
+    types: 'off',
+    unresolved: 'off',
+  },
+  workspaces: {
+    '.': {
+      ignoreDependencies: [
+        '@babel/code-frame',
+        '@babel/core',
+        '@babel/eslint-parser',
+        '@babel/parser',
+        '@babel/types',
+        '@nx/workspace',
+        'cross-fetch',
+        'glob',
+        'husky',
+        'jest-specific-snapshot',
+        'make-dir',
+        'ncp',
+        'tmp',
+
+        // imported in eslint.config.js
+        '@typescript-eslint/utils',
+        // imported in eslint.config.mjs
+        '@typescript-eslint/eslint-plugin-internal',
```

Comment:
ESLint actually does support `.mjs` now! We just never updated in our repo. `rm eslint.config.js` and these no longer need to be here.

```suggestion
```

Reply:
It looks like `eslint.config.mjs` is not currently supported by `@nx/eslint`:

```
Linting "repo-tools"...

 NX   No ESLint configuration found in /<...>/typescript-eslint/packages/repo-tools.


Error: No ESLint configuration found in /<...>/typescript-eslint/packages/repo-tools.
    at CascadingConfigArrayFactory._finalizeConfigArray (/<...>/typescript-eslint/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:4000:19)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/<...>/typescript-eslint/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3791:21)
    at FileEnumerator._iterateFilesRecursive (/<...>/typescript-eslint/node_modules/eslint/lib/cli-engine/file-enumerator.js:475:52)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (/<...>/typescript-eslint/node_modules/eslint/lib/cli-engine/file-enumerator.js:299:49)
    at iterateFiles.next (<anonymous>)
    at CLIEngine.executeOnFiles (/<...>/typescript-eslint/node_modules/eslint/lib/cli-engine/cli-engine.js:797:48)
    at ESLint.lintFiles (/<...>/typescript-eslint/node_modules/eslint/lib/eslint/eslint.js:551:23)
    at run (/<...>/typescript-eslint/node_modules/@nx/eslint/src/executors/lint/lint.impl.js:79:36)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

 NX   Running target lint for project repo-tools and 5 tasks it depends on failed

```

https://github.com/nrwl/nx/issues/22576

https://github.com/nrwl/nx/blob/f76b8c63c12ddfcb8de8d648cded026eec968ced/packages/eslint/src/executors/lint/lint.impl.ts#L46

Reply:
Ah, right